Make CI faster

Luthaf opened the pull request.

Luthaf committed:
Use libmamba to solve environements

Luthaf committed:
Sort examples when generating CI jobs

ceriottm commented on .github/workflows/docs.yml:
```diff
@@ -56,6 +56,7 @@ jobs:
         run: python -m pip install nox docutils
 
       - name: build example
+        timeout-minutes: 10
```
I'd make this 16' so it doesn't fail current examples ^_^

Luthaf replied:
I did 12min, pet-mad-nc takes 10min currently, so this should be enough of a buffer for run to run variations

pfebrer commented on src/get_examples.py:
```diff
@@ -67,7 +68,7 @@ def create_ci_jobs_json(modified_files: str):
         return ""
 
-    return json.dumps({"example-name": examples})
+    return json.dumps({"example-name": list(sorted(examples))})
```
Is this just so that the list of CI runs looks nicer? Or there was some deeper problem?

Luthaf replied:
Just to make the list nicer, and match the local ordering in my editor

Luthaf committed:
Force the use of CPU pytorch to reduce downloads

Luthaf committed:
Print the list of installed packages

Luthaf committed:
Add a 12 min timeout for building examples

Ideally the timeout should be smaller, but we have a couple
of examples that currently take 8 to 10min.

Luthaf committed:
Do not build the website when building one example on CI

pfebrer commented on .github/workflows/docs.yml:
```diff
@@ -61,7 +61,7 @@ jobs:
         run: |
           conda config --set notify_outdated_conda false
-          nox -e ${{ matrix.example-name }}
+          nox -e ${{ matrix.example-name }} -- --no-website
```
Why is this?

Luthaf replied:
When running `nox -e some-example` locally, we want to both generate the rst files from the python scripts, and to turn this rst into HTML for local visualization. On CI the second step is a separate CI job, so we can skip the rst => HTML transformation in this step and save some time.

pfebrer replied:
Ok, in principle this should also work with PRs, where for untouched exemples we download the artifacts from the latest build, no?

Luthaf replied:
Yes! The website itself is built below, after downloading all artifacts

Luthaf commented on .github/workflows/docs.yml:
```diff
@@ -100,7 +105,7 @@ jobs:
 
       - name: build documentation
-        run: nox -e build_docs
+        run: nox -e build_website
```
@pfebrer here

ceriottm reviewed (approved):
LGTM (and it IS faster)
